drivers/gpio/gpio_sn74hc595.c

```diff
@@ -21,8 +21,8 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(gpio_sn74hc595, CONFIG_GPIO_LOG_LEVEL);
 
-#if CONFIG_SPI_INIT_PRIORITY >= CONFIG_GPIO_SN74HC595_INIT_PRIORITY
-#error SPI_INIT_PRIORITY must be lower than SN74HC595_INIT_PRIORITY
+#if CONFIG_SPI_INIT_PRIORITY >= CONFIG_GPIO_SPI_INIT_PRIORITY
```

Comment:
This could be factored-out and put into common code now (sorry if that's already been done, i just didn't see it).

Reply:
Yeah was thinking about it wondering how to do it, my concern is that if I just dump it in the common gpio.h header it may end up breaking the build if someone is changing the i2c priority without changing the two new symbols even if they don't use any i2c/spi port expander. Alternatively I could put it in its own local header and include it only on drivers that needs it. What do you think?

Reply:
Probably the latter option. It is tricky though..

Reply:
I'll give it a try, see how ugly it gets :-)

Reply:
Added a patch for that, looks a bit hackish but I can't quite think of a clean solution. Could also add a hidden symbol to control that and then flag it in Kconfig but it'd probably be equally messy. Guess eventually we'll get dt based priority anyway and it'll clean all this mess.